app/src/main/java/ar/com/wolox/android/training/ui/login/LogInFragment.java

```diff
@@ -0,0 +1,93 @@
+package ar.com.wolox.android.training.ui.login;
+
+import android.content.Context;
+import android.content.Intent;
+import android.content.SharedPreferences;
+import android.os.Bundle;
+import android.text.method.LinkMovementMethod;
+import android.view.View;
+import android.widget.EditText;
+import android.widget.TextView;
+
+import ar.com.wolox.android.R;
+import ar.com.wolox.android.training.ui.home.HomeActivity;
+import ar.com.wolox.android.training.ui.signup.SignUpActivity;
+import ar.com.wolox.wolmo.core.fragment.WolmoFragment;
+import butterknife.BindView;
+import butterknife.OnClick;
+
+public class LogInFragment extends WolmoFragment<LogInPresenter> implements LogInView {
+
+    private static final String USER = "USER";
+    @BindView(R.id.fragment_login_signup_terms) TextView mTermsTxt;
+    @BindView(R.id.fragment_login_email_edit_text) EditText mEmailTxt;
+    @BindView(R.id.fragment_login_password_edit_text) EditText mPasswordTxt;
+
+    // Fragments default constructors shouldn't be overridden, always prefer this method instead
+    public static LogInFragment newInstance() {
+        Bundle args = new Bundle();
+        LogInFragment fragment = new LogInFragment();
+        fragment.setArguments(args);
+        return fragment;
+    }
+
+    @OnClick(R.id.fragment_login_login_button)
+        public void LogInValidate(){
+        mEmailTxt.setError(null);
+        mPasswordTxt.setError(null);
+            if (mEmailTxt.length() != 0 && mPasswordTxt.length() != 0) {
+                if (!android.util.Patterns.EMAIL_ADDRESS.matcher(mEmailTxt.getText()).matches()) {
```

Comment:
Try to make new methods for this like  !isValidEmail(mailText) . You nerver know when the implementation could change :3